### PR TITLE
feat(app): Provide group value defined in mapping in applications

### PR DESCRIPTION
**Pull request by FunKuchen**
## Type of this PR 

- Enhancement

## Description

Provide getGroup Method in OperatingSystem Interface and AbstractSimulationUnit. Adds a setGroup Method in AbstractSimulationUnit which is used to set the group value read from UnitRegistration message for each simulation object added to the simulation. This happens in each registerUnit method in the UnitSimulator class. That way one can access the group value in any application.

What is this PR about?

## Issue(s) related to this PR

Internal issue 284
  
## Affected parts of the online documentation

--
Changes in the documentation required?

## Definition of Done

- [x] You have read CONTRIBUTING.md carefully.
- [x] You have signed the [Contributor License Agreement](http://www.eclipse.org/legal/CLA.php).
- [x] Your GitHub user id is linked with your Eclipse Account.
- [x] `origin/main` has been merged into your Fork.
- [x] New functionality is covered by unit tests or integration tests. Code coverage must not decrease.
- [x] There are no new SpotBugs warnings. 
- [x] Coding guidelines have been followed (see CONTRIBUTING.md).
- [x] All tests pass.

### CODE REVIEW

File: fed/mosaic-application/src/main/java/org/eclipse/mosaic/fed/application/app/api/os/OperatingSystem.java [hunk 1]
```diff
@@ -143,4 +143,6 @@ default long getSimulationTimeMs() {
      * @return the list containing all applications.
      */
     <A extends Application> Iterable<A> getApplicationsIterator(Class<A> applicationClass);
+
+    String getGroup();
```
Comment (kschrab) [inline fed/mosaic-application/src/main/java/org/eclipse/mosaic/fed/application/app/api/os/OperatingSystem.java:147]: add javadoc

**Review by kschrab — CHANGES_REQUESTED** [review]
everything fine, still missing documentation of API method

**Comment by FunKuchen** [comment]
Is everything correct now?

File: fed/mosaic-application/src/main/java/org/eclipse/mosaic/fed/application/app/api/os/OperatingSystem.java (hunk 1 above)
Comment (FunKuchen) [reply fed/mosaic-application/src/main/java/org/eclipse/mosaic/fed/application/app/api/os/OperatingSystem.java:147]: Alright, will do.

File: fed/mosaic-application/src/main/java/org/eclipse/mosaic/fed/application/app/api/os/OperatingSystem.java [hunk 2]
```diff
@@ -143,4 +143,12 @@ default long getSimulationTimeMs() {
      * @return the list containing all applications.
      */
     <A extends Application> Iterable<A> getApplicationsIterator(Class<A> applicationClass);
+
+    /**
+     * Get the group of this simulation unit, defined in the mapping.
+     *
+     * @return the group of this simulation unit.
+     */
+
```
Comment (kschrab) [inline fed/mosaic-application/src/main/java/org/eclipse/mosaic/fed/application/app/api/os/OperatingSystem.java:152]: remove this empty line and we're good :)

**Review by kschrab — APPROVED** [review]

**Pull request merged**